src/Soil/Biogeochemistry/Biogeochemistry.jl

```diff
@@ -137,6 +158,311 @@ function DETECTModelParameters(;
+    boundary_conditions::BC 
+    "A tuple of sources, each of type AbstractSource"
+    sources::S
+    " Drivers"
+    driver::DT
+end
+
+
+"""
+DETECTModel{FT}(;
+        parameters::DETECTParameters{FT},
+        domain::ClimaLSM.AbstractDomain,
+        boundary_conditions::NamedTuple,
+        sources::Tuple,
+    ) where {FT}
+
+A constructor for `DETECTModel`.
+"""
+function DETECTModel{FT}(;
+    parameters::DETECTParameters{FT},
+    domain::ClimaLSM.AbstractDomain,
+    boundary_conditions::BC,
+    sources::Tuple,
+    drivers::DT
+) where {FT,BC, DT}
+    args = (parameters, domain, boundary_conditions, sources, drivers)
+    DETECTModel{FT, typeof.(args)...}(args...)
+end
+
+ClimaLSM.name(model::DETECTModel) = :DETECT;
+ClimaLSM.domain(model::DETECTModel) = :subsurface
+
+# 3. Prognostic and Auxiliary variables 
+
+ClimaLSM.prognostic_vars(::DETECTModel) = (:C,) # pCO2 in soil, [ppm] # stored in Y.DETECT.C
+ClimaLSM.prognostic_types(::DETECTModel{FT}) where {FT} = (FT,)
+ClimaLSM.auxiliary_vars(::DETECTModel) = (:D, :Sₘ, :Sᵣ) # Diffusivity, Source (microbe + root) # stored in p.DETECT.D
+ClimaLSM.auxiliary_types(::DETECTModel{FT}) where {FT} = (FT, FT, FT)
+
+
+# 4. RHS
+
+"""
+    make_rhs(model::DETECTModel)
+
+An extension of the function `make_rhs`, for the DETECT equation.
+This function creates and returns a function which computes the entire
+right hand side of the PDE for `C`, and updates `dY.soil.C` in place
+with that value.
+This has been written so as to work with Differential Equations.jl.
+"""
+function ClimaLSM.make_rhs(model::DETECTModel)
+    function rhs!(dY, Y, p, t)
+        z = ClimaCore.Fields.coordinate_field(model.domain.space).z
+        Δz_top, Δz_bottom = get_Δz(z)
+
+        top_flux_bc = boundary_flux(
+            model.boundary_conditions.CO2.top,
+            TopBoundary(),
+            Δz_top,
+            Y,
+            p,
+            t,
+        )
+        bot_flux_bc = boundary_flux(
+            model.boundary_conditions.CO2.bottom,
+            BottomBoundary(),
+            Δz_bottom,
+            Y,
+            p,
+            t,
+        )
+
+        interpc2f = ClimaCore.Operators.InterpolateC2F()
+        gradc2f_C = ClimaCore.Operators.GradientC2F() # set a BC on state
+        divf2c_C = ClimaCore.Operators.DivergenceF2C(
+            top = ClimaCore.Operators.SetValue(ClimaCore.Geometry.WVector.(top_flux_bc)),
+            bottom = ClimaCore.Operators.SetValue(ClimaCore.Geometry.WVector.(bot_flux_bc)),
+        ) # -∇ ⋅ (-D∇C), where -D∇C is a flux of C02. ∇C point in direction of increasing C, so the flux is - this.
+        @. dY.DETECT.C =
+	                 -divf2c_C(-interpc2f(p.DETECT.D)*gradc2f_C(Y.DETECT.C))
+
+        # Source terms are added in here
+        for src in model.sources
+            source!(dY, src, Y, p, model.parameters)
+        end
+
+    end
+    return rhs!
+end
+
+abstract type AbstractCarbonSource end
```

Comment:
docs strings for these :)

Reply:
I wrote something but I am not sure it is a good doc strings.